nova_vm/src/ecmascript/execution/agent.rs

```diff
@@ -189,6 +193,10 @@ impl Agent {
     pub(crate) fn running_execution_context_mut(&mut self) -> &mut ExecutionContext {
         self.execution_context_stack.last_mut().unwrap()
     }
+
+    pub fn get_hoost_hooks(&self) -> &'static dyn HostHooks {
```

Comment:
**nitpick (blocking):** It's probably not a good idea to expose the entire host hooks through the Agent. eg. It probably shouldn't be possible to call the uncaught rejection handler.

So maybe just do a `get_host_storage` or something?